pkg/agent/apiserver/handlers/multicast/handler_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicast
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/multicast"
+	"antrea.io/antrea/pkg/features"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	queriertest "antrea.io/antrea/pkg/querier/testing"
+)
+
+func TestPodMulticastStatsQuery(t *testing.T) {
+	features.DefaultMutableFeatureGate.Set("Multicast=true")
+	defer features.DefaultMutableFeatureGate.Set("Multicast=false")
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	testcases := map[string]struct {
+		name                   string
+		namespace              string
+		expectedStatus         int
+		expectedContent        []Response
+		getPodStatsResult      *multicast.PodTrafficStats
+		gettAllPodsStatsResult map[*interfacestore.InterfaceConfig]*multicast.PodTrafficStats
+	}{
+		"Hit PodMulticastStats query, namespace provided": {
+			name:           "pod1",
+			namespace:      "namespaceA",
+			expectedStatus: http.StatusOK,
+			expectedContent: []Response{
+				{
+					PodName:      "pod1",
+					PodNamespace: "namespaceA",
+					Inbound:      "22",
+					Outbound:     "33",
+				},
+			},
+			getPodStatsResult: &multicast.PodTrafficStats{Inbound: 22, Outbound: 33},
+		},
+		"Miss PodMulticastStats query, namespace and name provided": {
+			name:              "pod1",
+			namespace:         "namespaceA",
+			expectedStatus:    http.StatusNotFound,
+			getPodStatsResult: nil,
+		},
+		"Namespace not provided": {
+			name:           "pod1",
+			namespace:      "",
+			expectedStatus: http.StatusServiceUnavailable,
+		},
+		"Name not provided": {
```

Comment:
should we return all stats in the namespace if only namespace is specified? Not strong opinion.

Reply:
Updated. I agree with you. this command will follow the style of `antctl get podinterface`.